src/hotspot/share/utilities/compilerWarnings_gcc.hpp

```diff
@@ -63,4 +63,56 @@
 
 #endif // clang/gcc version check
 
+#if __GNUC__ >= 9
+
+// AIX also needs a 64 bit NULL to work as a null address pointer.
+// Most system includes on AIX would define it as an int 0 if not already defined with one
+// exception: /usr/include/dirent.h will unconditionally redefine NULL to int 0 again.
+// In this case you need to copy the following defines to a position after #include <dirent.h>
+#include <dirent.h>
+#ifdef _LP64
+  #undef NULL
+  #define NULL 0L
+#else
+  #ifndef NULL
+    #define NULL 0
+  #endif
+#endif
+
+#include <sys/socket.h>
+#include <stdio.h>
+
+#define FORBID_C_FUNCTION(signature, alternative) \
+  extern "C" signature __attribute__((__warning__(alternative)))
+
+#define PRAGMA_PERMIT_FORBIDDEN_C_FUNCTION(name) \
+  PRAGMA_DISABLE_GCC_WARNING("-Wattribute-warning")
+
+FORBID_C_FUNCTION(int      closedir(DIR*),                            "use os::closedir");
+FORBID_C_FUNCTION(int      connect(int, const struct sockaddr*, socklen_t), "use os::connect");
+FORBID_C_FUNCTION(FILE*    fdopen(int, const char*),                  "use os::fdopen");
+FORBID_C_FUNCTION(void     flockfile(FILE*),                          "use os::flockfile");
+FORBID_C_FUNCTION(FILE*    fopen(const char*, const char*),           "use os::fopen");
+FORBID_C_FUNCTION(int      fsync(int),                                "use os::fsync");
+FORBID_C_FUNCTION(int      ftruncate(int, off_t),                     "use os::ftruncate");
```

Comment:
Shouldn't this be ftruncate for BSD and ftruncate64 for other Posix (not sure what Windows has)?

Reply:
Platform agnostic code would call ftruncate(), not ftruncate64().  So I think this is correct as is.

Reply:
You need to enable the warning for the function that we would use, which we are not supposed to use and that would be `ftruncate64` on Linux.